data/json/items/generic.json

```diff
@@ -2339,47 +2339,68 @@
         "volume" : 0,
         "color" : "white",
         "symbol" : "!",
-        "to_hit" : -2,
-        "cutting" : -1,
-        "bashing" : -1,
         "description" : "Smoke swirls from the lit end of this cigarette, filling the air with the thrilling smell of burning chemicals.",
-        "price" : 0,
         "category": "drugs",
         "flags" : ["LITCIG", "TOBACCO", "LIGHT_4"]
     },
     {
-        "type" : "GENERIC",
+        "type" : "COMESTIBLE",
         "id" : "cig_butt",
         "name" : "cigarette butt",
+        "use_action" : "SMOKING",
         "weight" : 0,
+        "comestible_type" : "MED",
         "color" : "dark_gray",
         "symbol" : "!",
         "description" : "What was once a wonderfully addictive tube of dried tobacco leaf is now just a smelly piece of trash.  What a tragedy!\nThe leftover tobacco in a few of these could probably be used to roll another cigarette.  If you're willing to go that far...",
-        "price" : 0,
         "volume" : 0,
         "category": "other"
     },
     {
         "type" : "GENERIC",
+        "id" : "cig_butt_lit",
+        "name" : "cigarette butt",
+        "weight" : "0",
+        "volume" : "0",
+        "color" : "dark_gray",
+        "symbol" : "!",
+        "description" : "Smelly piece of trash. You must be very desperate if you even try to smoke it.",
+        "category" : "drugs",
+        "flags" : ["TABACCO","LITCIG","LIGHT_2"]
+
+    },
+    {
+        "type" : "GENERIC",
+        "id" : "joint_roach_lit",
+        "name" : "joint roach",
+        "name_plural": "joint roaches",
+        "color" : "green",
+        "symbol" : "!",
+        "description" : "The smoked-down butt of a joint, a reminder of some good times.  Why you smoke it again ?.",
```

Comment:
Excess space before question mark. Also, excess period after question mark.
Also, "Why you smoke it again?" is unfinished sentence. Maybe "Why do you want to smoke it again?"